packages/sit-onyx/src/components/OnyxFlyoutMenu/future/OnyxFlyoutMenu.vue

```diff
@@ -0,0 +1,123 @@
+<script setup lang="ts" generic="TValue extends SelectOptionValue = SelectOptionValue">
+import { createMenuButton } from "@sit-onyx/headless";
+import type { SelectOptionValue } from "../../../types";
+import { computed, ref, type VNode } from "vue";
+
+// const props = defineProps<OnyxFlyoutMenuProps<TValue>>();
+
+const slots = defineSlots<{
+  /**
+   * The trigger for the flyout menu
+   */
+  default(): VNode[];
+  /**
+   * OnyxListItems to show
+   */
+  options?(): VNode[];
+  /**
+   * Optional header content to display above the options.
+   */
+  header?(): unknown;
+  /**
+   * Optional footer content to display below the options (will replace `message` property).
+   */
+  footer?(): unknown;
+}>();
+
+const activeItem = ref<string>();
+
+const {
+  elements: { button, menu, menuItem, listItem, parentComponent },
+  state: { isExpanded },
+} = createMenuButton({
+  onSelect: (href) => {
+    activeItem.value = href;
+  },
```

Comment:
```suggestion
  onSelect: (value) => {
    activeItem.value = value;
  },
```

It could also be something other than a href right?